drivers/volume/portworx/portworx.go

```diff
@@ -4151,6 +4153,27 @@ func (p *portworx) verifyPortworxPv(pv *v1.PersistentVolume) error {
 	return err
 }
 
+func (p *portworx) GetVolumeIDFromPVC(pvcName string, namespace string) (string, error) {
+	pvc, err := core.Instance().GetPersistentVolumeClaim(pvcName, namespace)
+	if err != nil {
+		return "", fmt.Errorf("error getting PV name for PVC (%v/%v): %w", pvc.Namespace, pvc.Name, err)
+	}
+	pvName := pvc.Spec.VolumeName
+
+	pv, err := core.Instance().GetPersistentVolume(pvName)
+	if err != nil {
+		return "", fmt.Errorf("error getting PV %v: %w", pvName, err)
+	}
+
+	id, err := p.getVolumeIDFromPV(pv)
+	if err != nil {
+		return "", fmt.Errorf("error getting volume ID for PV %v: %w", pvName, err)
+
+	}
+
+	return id, err
```

Comment:
We can avoid this since Portworx volumes can be queried using the PV name. Is there any reason we need the internal volume ID here?

Reply:
Will change it to use PV name. Will take care of this in the next revision.